src/pages/SignUp.js

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from "react";
+import { Link, useHistory } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { signUpThunkCreator } from "../store/user/actions";
+import { showMessageThunkCreator } from "../store/appState/actions";
+
+import Form from "react-bootstrap/Form";
+import Container from "react-bootstrap/Container";
+import Button from "react-bootstrap/Button";
+import { Col } from "react-bootstrap";
+
+export default function SignUp() {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [role, setRole] = useState("");
+  const [validated, setValidated] = useState(false);
+
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  function submitForm(event) {
+    event.preventDefault();
+
+    if (password !== confirmPassword) {
+      dispatch(showMessageThunkCreator("Passwords do not match", "danger"));
+      console.log("Password don't match");
+    } else {
+      const form = event.currentTarget;
+      if (form.checkValidity() === false) {
+        event.stopPropagation();
+      } else {
+        setValidated(true);
+        dispatch(signUpThunkCreator({ name, email, password, role }));
+        setName("");
+        setEmail("");
+        setPassword("");
+        setConfirmPassword("");
+        setRole("");
+        history.push("/");
```

Comment:
mmm you redirect before knowing if the submit worked? why dont you pass down the `history` object to the thunkCreator and do the history.push after you get the success back from the server?